Healthy/Classes/Modules/Splash/DemoViewController.swift

```diff
@@ -0,0 +1,225 @@
+        
+        pages.append(page1)
+        pages.append(page2)
+        pages.append(page3)
+//        pages.append(page4)
+        
+        setViewControllers([pages[initialPage]], direction: .forward, animated: true, completion: nil)
+    }
+    
+    func style() {
+        pageControl.translatesAutoresizingMaskIntoConstraints = false
+        pageControl.currentPageIndicatorTintColor = .black
+        pageControl.pageIndicatorTintColor = .systemGray2
+        pageControl.numberOfPages = pages.count
+        pageControl.currentPage = initialPage
+
+        skipButton.translatesAutoresizingMaskIntoConstraints = false
+        skipButton.setTitleColor(.systemBlue, for: .normal)
+        skipButton.setTitle("Skip", for: .normal)
+        skipButton.addTarget(self, action: #selector(skipTapped(_:)), for: .primaryActionTriggered)
+
+        nextButton.translatesAutoresizingMaskIntoConstraints = false
+        nextButton.setTitleColor(.systemBlue, for: .normal)
+        nextButton.setTitle("Next", for: .normal)
+        nextButton.addTarget(self, action: #selector(nextTapped(_:)), for: .primaryActionTriggered)
+    }
+    
+    func layout() {
+        view.addSubview(pageControl)
+        view.addSubview(nextButton)
+        view.addSubview(skipButton)
+        
+        NSLayoutConstraint.activate([
+            pageControl.widthAnchor.constraint(equalTo: view.widthAnchor),
+            pageControl.heightAnchor.constraint(equalToConstant: 20),
+            pageControl.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+
+            
+            skipButton.leadingAnchor.constraint(equalToSystemSpacingAfter: view.leadingAnchor, multiplier: 2),
+
+            
+            view.trailingAnchor.constraint(equalToSystemSpacingAfter: nextButton.trailingAnchor, multiplier: 2),
+        ])
+        
+        // for animations
+        pageControlBottomAnchor = view.bottomAnchor.constraint(equalToSystemSpacingBelow: pageControl.bottomAnchor, multiplier: 2)
+        skipButtonTopAnchor = skipButton.topAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.topAnchor, multiplier: 2)
+        nextButtonTopAnchor = nextButton.topAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.topAnchor, multiplier: 2)
+        
+        pageControlBottomAnchor?.isActive = true
+        skipButtonTopAnchor?.isActive = true
+        nextButtonTopAnchor?.isActive = true
+    }
+}
+
+// MARK: - DataSource
+
+extension DemoViewController: UIPageViewControllerDataSource {
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+
+        guard let currentIndex = pages.firstIndex(of: viewController) else { return nil }
+        
+        if currentIndex == 0 {
+            return pages.last               // wrap last
+        } else {
+            return pages[currentIndex - 1]  // go previous
+        }
```

Comment:
We should move any type of logic to viewModel
Note that to make it later using `Combine`

Reply:
📎📍📝